src/overlays/effects/ovl_Effect_Ss_En_Ice/z_eff_ss_en_ice.c

```diff
@@ -1,30 +1,157 @@
 
-s32 EffectSsEnIce_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
-void func_8097F62C(PlayState* play, u32 index, EffectSs* this);
-void func_8097F7EC(PlayState* play, u32 index, EffectSs* this);
+u32 EffectSsEnIce_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+void EffectSsEnIce_UpdateFlying(PlayState* play, u32 index, EffectSs* this);
+void EffectSsEnIce_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsEnIce_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_En_Ice_InitVars = {
     EFFECT_SS_EN_ICE,
     EffectSsEnIce_Init,
 };
 
-#endif
+u32 EffectSsEnIce_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsEnIceInitParams* initParams = PARAMS;
+
+    if (initParams->type == ENICE_TYPE_FLYING) {
+        Math_Vec3f_Copy(&this->pos, &initParams->pos);
+
+        if (initParams->actor != NULL) {
+            Math_Vec3f_Diff(&this->pos, &initParams->actor->world.pos, &this->vec);
+        } else {
+            Math_Vec3f_Copy(&this->vec, &initParams->pos);
+        }
+
+        Math_Vec3f_Copy(&this->velocity, &gZeroVec3f);
+        Math_Vec3f_Copy(&this->accel, &gZeroVec3f);
+        this->life = 10;
+        this->actor = initParams->actor;
+        this->draw = EffectSsEnIce_Draw;
+        this->update = EffectSsEnIce_UpdateFlying;
+        this->rScale = initParams->scale * 100.0f;
+        this->rPrimColorR = initParams->primColor.r;
+        this->rPrimColorG = initParams->primColor.g;
+        this->rPrimColorB = initParams->primColor.b;
+        this->rPrimColorA = initParams->primColor.a;
+        this->rEnvColorR = initParams->envColor.r;
+        this->rEnvColorG = initParams->envColor.g;
+        this->rEnvColorB = initParams->envColor.b;
+        this->rAlphaMode = 1;
+        this->rPitch = randPlusMinusPoint5Scaled(0x10000);
+    } else if (initParams->type == ENICE_TYPE_NORMAL) {
+        Math_Vec3f_Copy(&this->pos, &initParams->pos);
+        Math_Vec3f_Copy(&this->vec, &initParams->pos);
+        Math_Vec3f_Copy(&this->velocity, &initParams->velocity);
+        Math_Vec3f_Copy(&this->accel, &initParams->accel);
+        this->life = initParams->life;
+        this->draw = EffectSsEnIce_Draw;
+        this->update = EffectSsEnIce_Update;
+        this->rLifespan = initParams->life;
+        this->rScale = initParams->scale * 100.0f;
+        this->rYaw = Math_FAtan2F(initParams->velocity.z, initParams->velocity.x);
+        this->rPitch = 0;
+        this->rPrimColorR = initParams->primColor.r;
+        this->rPrimColorG = initParams->primColor.g;
+        this->rPrimColorB = initParams->primColor.b;
+        this->rPrimColorA = initParams->primColor.a;
+        this->rEnvColorR = initParams->envColor.r;
+        this->rEnvColorG = initParams->envColor.g;
+        this->rEnvColorB = initParams->envColor.b;
+        this->rAlphaMode = 0;
+    } else {
+        return 0;
+    }
+
+    return 1;
+}
```

Comment:
Ooh, one that's actually not always true. Don't suppose it matters?

Reply:
Actually, effect_soft_sprite has a failsafe, where if the init function of an effect returns 0 (false) then it resets the entry.

Reply:
Ah, I see, that's where the true/false comes into play.

Reply:
Yeah, I've only ever seen `0/1` in both OoT and MM, and since it is part of an if conditional, it makes sense to be `true/false`. Again just haven't got around to it. Will probably do a global cleanup of all effects after they all get in.